### PR TITLE
Adding design genotyping primers to the view_experiment view

### PR DESCRIPTION
**Pull request by Jana-A**

### CODE REVIEW

**Comment by Jana-A** [comment]
@jkent20 This is ready for review.
Test link: http://www.sanger.ac.uk/htgt/lims2/user/view_experiment/?experiment_id=230
Redmine ticket number is 12411.

File: lib/LIMS2/Model/Plugin/Project.pm [hunk 1]
```diff
@@ -327,6 +327,22 @@ sub retrieve_experiment{
     return $experiment;
 }
 
+sub retrieve_genotyping{
+    my ($self, $my_design_id) = @_;
+    my %search_cond = ( design_id => $my_design_id );
+    my @records = $self->schema->resultset('GenotypingPrimer')->search( \%search_cond )->all;
```
Comment (jkent20) [inline lib/LIMS2/Model/Plugin/Project.pm]: Calls DBIx resultset, seems completely fine, however there is a method in lib/LIMS2/Model.pm called 'retrieve'. Simple method, just does a query based on parameters with a few checks, might be worth looking at. there is also a method in the same place called 'retrieve_list' does the same thing, just allows multiple rows returned

Comment (Jana-A) [inline lib/LIMS2/Model/Plugin/Project.pm]: @jkent20 The retrieve method seems to be able to return only 1 object/row (see https://github.com/htgt/LIMS2-WebApp/blob/devel/lib/LIMS2/Model.pm#L208)
In this experiment the genotyping data is 4 rows.

Comment (jkent20) [inline lib/LIMS2/Model/Plugin/Project.pm]: @Jana-A Yes, the retrieve method only returns 1 row, however there is a similar method called 'retrieve_list' which should be able to return multiple rows

File: lib/LIMS2/Model/Plugin/Project.pm [hunk 2]
```diff
@@ -327,6 +327,20 @@ sub retrieve_experiment{
     return $experiment;
 }
 
+sub retrieve_genotyping{
+    my ($self, $param) = @_;
+    my $records = $self->retrieve_list('GenotypingPrimer', $param);
+    my $count = 0;
+    my $res;
+    foreach (@{$records})
```
Comment (jkent20) [inline lib/LIMS2/Model/Plugin/Project.pm]: perhaps might be better to be
    `foreach my $record (@{$records}) {`

This way is slightly more verbose but allows for quicker understanding on what is contained in the local variable `$_`

File: lib/LIMS2/Model/Plugin/Project.pm [hunk 3]
```diff
@@ -327,6 +327,20 @@ sub retrieve_experiment{
...
+    {
+        $res->[$count]->{sequence} = $_->{_column_data}->{seq};
```
Comment (jkent20) [inline lib/LIMS2/Model/Plugin/Project.pm]: Then here would be:
    `$res->[$count]->{sequence} = $record->{_column_data}->{seq};`

File: lib/LIMS2/Model/Plugin/Project.pm [hunk 4]
```diff
@@ -327,6 +327,20 @@ sub retrieve_experiment{
...
+        $res->[$count]->{type} = $_->{_column_data}->{genotyping_primer_type_id};
```
Comment (jkent20) [inline lib/LIMS2/Model/Plugin/Project.pm]: And here:
    `$res->[$count]->{type} = $record->{_column_data}->{genotyping_primer_type_id};`

**Pull request merged**